From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qeth: add missing wake_up call
Patch-mainline: yes
References: bnc#752484,LTC#79899

Symptom:     cclengine hangs when running Communications Controller
             for Linux on System z
Problem:     To send commands on the write channel 8 buffers exist.
             If all 8 buffers are used, a wait is triggered on the
             write channel. When such buffers are freed, a wake_up
             is needed.
Solution:    This patch adds the missing wake_up in
             qeth_release_buffer(). The fix is especially important
             when running Communications Controller for Linux on
             System z.

Acked-by: John Jolly <jjolly@suse.de>

---
 drivers/s390/net/qeth_core_main.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/s390/net/qeth_core_main.c
+++ b/drivers/s390/net/qeth_core_main.c
@@ -678,6 +678,7 @@ void qeth_release_buffer(struct qeth_cha
 	iob->callback = qeth_send_control_data_cb;
 	iob->rc = 0;
 	spin_unlock_irqrestore(&channel->iob_lock, flags);
+	wake_up(&channel->wait_q);
 }
 EXPORT_SYMBOL_GPL(qeth_release_buffer);
 
